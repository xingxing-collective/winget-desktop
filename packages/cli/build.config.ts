import { defineBuildConfig } from 'unbuild'

export default defineBuildConfig({
  rollup: {
    dts: {
      respectExternal: false,
    },
  },
  entries: ['src/index']
})