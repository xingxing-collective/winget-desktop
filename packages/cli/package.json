{
  "name": "@winget-desktop/kit",
  "version": "1.0.0",
  "private": false,
  "engines": {
    "node": "^16.10.0 || >=18.0.0"
  },
  "scripts": {
    "build": "unbuild",
    "dev:prepare": "unbuild --stub",
    "prepack": "unbuild"
  },
  "packageManager": "pnpm@10.9.0",
  "dependencies": {
    "jiti": "^2.4.2",
    "unbuild": "^3.5.0"
  }
}