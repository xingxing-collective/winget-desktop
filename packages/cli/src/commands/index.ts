export * from './powershell'
export * from './winget'