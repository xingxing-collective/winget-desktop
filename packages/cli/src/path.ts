import { resolve } from "node:path";

export const proRoot = resolve(__dirname, '..', '..', '..')

export const pkgRoot = resolve(proRoot, 'packages')

export const appRoot = resolve(proRoot, '.winget-desktop')

export const appsPath = resolve(appRoot, 'apps.json')

export const settingsPath = resolve(appRoot, 'settings.json')