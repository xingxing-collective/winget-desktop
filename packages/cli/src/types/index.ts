export type PackageFieldMatchOption = 'Equals' | 'EqualsCaseInsensitive' | 'StartsWithCaseInsensitive' | 'ContainsCaseInsensitive'
