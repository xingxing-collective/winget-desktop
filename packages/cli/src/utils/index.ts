export * from './storage'
export * from './formatter'
export * from './settings'