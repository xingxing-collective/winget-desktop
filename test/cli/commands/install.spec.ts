import { describe, expect, it } from "vitest";
import { install } from "../../../packages/cli/src";

describe('Install-WinGetPackage', () => {

  it('Install a package using a query', async () => {
    const result = await install('Microsoft.PowerShell')
    expect(result).toMatchSnapshot()
  }, Infinity)

  it('Install a package by Id', async () => {
    const result = await install([['-Id', 'Microsoft.PowerShell']])
    expect(result).toMatchSnapshot()
  }, Infinity)

  it('Install a package by Name', async () => {
    const result = await install([['-Name', 'PowerToys']])
    expect(result).toMatchSnapshot()
  }, Infinity)

  it('Install a specific version of a package', async () => {
    const result = await install([['Microsoft.PowerShell'], ['-Version', '7.4.4.0']])
    expect(result).toMatchSnapshot()
  }, Infinity)

})