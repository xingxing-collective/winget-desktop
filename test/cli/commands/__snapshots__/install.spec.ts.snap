// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Install-WinGetPackage > Install a package by Id 1`] = `
{
  "CorrelationData": "",
  "ExtendedErrorCode": null,
  "Id": "Microsoft.PowerShell",
  "InstallerErrorCode": 0,
  "Name": "PowerShell",
  "RebootRequired": false,
  "Source": "winget",
  "Status": "Ok",
}
`;

exports[`Install-WinGetPackage > Install a package using a query 1`] = `
{
  "CorrelationData": "",
  "ExtendedErrorCode": null,
  "Id": "Microsoft.PowerShell",
  "InstallerErrorCode": 0,
  "Name": "PowerShell",
  "RebootRequired": false,
  "Source": "winget",
  "Status": "Ok",
}
`;

exports[`Install-WinGetPackage > Install a specific version of a package 1`] = `
{
  "CorrelationData": "",
  "ExtendedErrorCode": {
    "Data": {},
    "HResult": -2147012865,
    "HelpLink": null,
    "InnerException": null,
    "Message": "异常来自 HRESULT:0x80072EFF",
    "Source": null,
    "StackTrace": null,
    "TargetSite": null,
  },
  "Id": "Microsoft.PowerShell",
  "InstallerErrorCode": 0,
  "Name": "PowerShell",
  "RebootRequired": false,
  "Source": "winget",
  "Status": "InternalError",
}
`;
